src/builder.rs

```diff
@@ -44,7 +44,9 @@ impl<K: EnrKey> EnrBuilder<K> {
 
     /// Adds an arbitrary key-value to the `ENRBuilder`.
     pub fn add_value<T: Encodable>(&mut self, key: impl AsRef<[u8]>, value: &T) -> &mut Self {
-        self.add_value_rlp(key, rlp::encode(value).freeze())
+        let mut out = BytesMut::new();
+        value.encode(&mut out);
+        self.add_value_rlp(key, Bytes::copy_from_slice(&out))
```

Comment:
No need to copy again
```suggestion
        self.add_value_rlp(key, out.freeze())
```